src/main/java/seedu/address/logic/parser/inventory/InventoryAddIngredientCommandParser.java

```diff
@@ -14,8 +30,53 @@
      * and returns a InventoryAddCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
-    public InventoryCommand parse(String args) throws ParseException {
-        throw new ParseException("Not implemented yet");
+    public InventoryAddIngredientCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        if (containsIngredient(args)) {
+            return parseAddIngredient(args);
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RecipeAddCommand.MESSAGE_USAGE));
+        }
     }
 
+    /**
+     * Parses the given {@code String} of arguments in the context of the RecipeAddIngredientCommand
+     * and returns a RecipeAddIngredientCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    private InventoryAddIngredientCommand parseAddIngredient(String args) throws ParseException {
```

Comment:
So instead, the code in the method can be moved directly into the `parse()` method.